### DISCUSSION TITLE
深入了解 ebpf map

### DISCUSSION THREAD

**Discussion by giscus** [edony-ink/comments#15] [General]
# 深入了解 ebpf map

探究 ebpf map 在 kernel 中加载和使用的原理，最后给出两个比较常见的 map 使用的示例

https://www.edony.ink/deepinsight-of-ebpf-map/

**Comment by hinzer** [comment]
请教一个问题。基于 ebpf map 机制能干哪些事情？掌握它测试内核会不会很方便啊，它和 ftrace 相比怎么样呢？

**Reply by edonyzpc** [reply]
ebpf map 主要是 ebpf 程序在内核态和用户态共享数据的。ebpf 实现了内核可编程能力，可以 tracing、阻断 syscall 等等，这其中如果涉及到数据交互（例如用户态配置 ebpf 程序等）就需要用到 map，另外 tracing kernel 的时候可以将 kernel struct 数据通过 map 输出到用户态。
tracing 领域对比的话，ebpf 跟 ftrace 相比更多的是可编程能力，ftrace 只能 trace 一些预定义好的 tracepoint
